### PR TITLE
Valset: Implement ListJailedValidators query

### PR DESCRIPTION
**Pull request by ueco-jb**
closes https://github.com/confio/poe-contracts/issues/87

### CODE REVIEW

File: contracts/tgrade-valset/src/contract.rs [hunk 1]
```diff
@@ -523,6 +529,45 @@ fn list_active_validators(
...
+    let start = start_after.map(|addr| Bound::exclusive(addr.as_str()));
+
+    let validators: Vec<OperatorResponse> = operators()
+        .range(deps.storage, start, None, Order::Ascending)
+        .map(|r| {
+            let (operator, info) = r?;
+
+            let jailed_until = JAIL
+                .may_load(deps.storage, &Addr::unchecked(&operator))?
+                .filter(|expires| !(cfg.auto_unjail && expires.is_expired(&env.block)));
```
Comment (maurolacy) [inline contracts/tgrade-valset/src/contract.rs:550]: I would reverse these nested iterators. First range over `JAIL` and then, if jailed, load the required data from `operators()`.

The reason being that at any given time, there will be far fewer jailed than operators.

File: contracts/tgrade-valset/src/error.rs [hunk 2]
```diff
@@ -48,6 +48,9 @@ pub enum ContractError {
     #[error("No validators")]
     NoValidators {},
 
+    #[error("Validator not jailed")]
+    ValidatorNotJailed {},
```
Comment (maurolacy) [inline contracts/tgrade-valset/src/error.rs:52]: No usages found in all places.

File: contracts/tgrade-valset/src/contract.rs [hunk 3]
```diff
@@ -523,6 +529,45 @@ fn list_active_validators(
...
+            match jailed_until {
+                Some(jailed_into) => Ok(Some(OperatorResponse {
+                    operator: operator.into(),
+                    metadata: info.metadata,
+                    pubkey: info.pubkey.into(),
+                    jailed_until: Some(jailed_into),
+                    active_validator: info.active_validator,
+                })),
+                None => Ok(None),
```
Comment (maurolacy) [inline contracts/tgrade-valset/src/contract.rs:559]: Perhaps a `JailedResponse`, with only `operator` and `jailed_until`? That will save the extra load from the storage (pun not intended).

Comment (ueco-jb) [reply contracts/tgrade-valset/src/contract.rs:559]: Actually `moniker` from metadata is required as well. I guess I could cut `pubkey` and `active_validator`, but would that still save the load?

File: contracts/tgrade-valset/src/error.rs (hunk 2 above)
Comment (ueco-jb) [reply contracts/tgrade-valset/src/error.rs:52]: Leftover.

File: contracts/tgrade-valset/src/contract.rs (hunk 3 above)
Comment (maurolacy) [reply contracts/tgrade-valset/src/contract.rs:559]: Why is `moniker` required? I would just send the operator address and `jailed_until`. The client can always enrich that if it needs / wants.

Comment (ueco-jb) [reply contracts/tgrade-valset/src/contract.rs]: Consensus is to include that info in response.

**Review by maurolacy — APPROVED** [review]
LGTM. Only issue I have is with sending metadata info as well.

Approving anyway, as this is what we agreed to do.

**Pull request merged**